### DISCUSSION TITLE
Deploy production API and frontend weekly

### DISCUSSION THREAD

**Discussion by sarayourfriend** [WordPress/openverse#1229] [Ideas]
We used to follow this procedure for the API and frontend, but since switching to ECS for both, it seems like this practice has fallen to the wayside. However, it is an important practice to avoid production deployments that are too big. Big deployments are more difficult to debug when they have issues because there are more changes that could cause problems. Production issues are not always obviously tied to a specific change. Furthermore, deploying too many changes at once can make rollbacks more difficult to manage.

I don't have any good ideas for how to ensure this happens other than adding them as MSR responsibilities for Monday. Deployments are easy to re-run, so even if a change that we needed to get in sooner is merged on Tuesday, it is trivial to re-run the production deployment again later in the week.

**Comment by zackkrida** [comment]
Big 👍 to bringing these back. I actually like the idea of making it an MSR thing, rather than having MSR go through all the rigamarole of outreach to deployers, which the past several weeks didn't even necessarily lead to deployments. Since we basically have 1-click deploys and 1-click rollbacks, it isn't much to ask of the MSR lead. 

That said, we'll definitely want to have the [Release app](https://github.com/WordPress/openverse/actions/workflows/release-app.yml) workflow working and documented. I think it's only waiting on #1193.

**Reply by AetherUnbound** [reply]
Agreed, deployments should now be pretty straightforward for any of the maintainers to execute! Having the MSR take care of it seems suitable, as long as we have clear instructions for how to roll back (which I believe we do).

**Comment by sarayourfriend** [comment]
I'm closing this. I've updated the MSR checklist to include frontend and API deployments on Mondays.
